python/ray/_private/thirdparty/pyamdsmi/pyamdsmi.py

```diff
@@ -0,0 +1,552 @@
+    RSMI_IOLINK_TYPE_XGMI           = 11
+    RSMI_IOLINK_TYPE_XGOP           = 12
+    RSMI_IOLINK_TYPE_GZ             = 13
+    RSMI_IOLINK_TYPE_ETHERNET_RDMA  = 14
+    RSMI_IOLINK_TYPE_RDMA_OTHER     = 15
+    RSMI_IOLINK_TYPE_OTHER          = 16
+    RSMI_IOLINK_TYPE_NUMIOLINKTYPES = 17
+    RSMI_IOLINK_TYPE_SIZE           = 0xFFFFFFFF
+
+
+## Library loading
+rocm_lib = None
+lib_load_lock = threading.Lock()
+_rocm_lib_refcount = 0
+
+
+## Function access, to prevent lib_load_lock deadlock
+_rocml_get_function_ptr_cache = dict()
+
+def _rocml_get_function_ptr(name):
+    global rocm_lib
+
+    if name in _rocml_get_function_ptr_cache:
+        return _rocml_get_function_ptr_cache[name]
+
+    lib_load_lock.acquire()
+    try:
+        # ensure library was loaded
+        if rocm_lib == None:
+            raise ROCMLError_Uninitialized
+        try:
+            _rocml_get_function_ptr_cache[name] = getattr(rocm_lib, name)
+            return _rocml_get_function_ptr_cache[name]
+        except AttributeError:
+            raise ROCMLError_FunctionNotFound
+    finally:
+        # lock is always freed
+        lib_load_lock.release()
+
+
+def _load_rocm_library():
+    """Load ROCm library if not already loaded"""
+    global rocm_lib
+
+    if rocm_lib == None:
+
+        lib_load_lock.acquire()
+
+        try:
+            if rocm_lib == None:
+                try:
+                    if sys.platform[:3] == 'win':
+                        raise ROCMLError_NotSupported('Windows platform is not supported yet')
+                    else:
+                        # assume linux
+                        path_librocm = _find_lib_rocm()
+                        cdll.LoadLibrary(path_librocm)
+                        rocm_lib = CDLL(path_librocm)
+                except OSError:
+                    raise ROCMLError_LibraryNotFound('ROCm library not found')
+                if rocm_lib == None:
+                    raise ROCMLError_LibraryNotFound('ROCm library not found')
+        finally:
+            lib_load_lock.release()
+
+
+def _find_lib_rocm():
+    """search for librocm and returns path
+    if search fails, returns empty string
+    """
+    rocm_path = os.environ.get('ROCM_PATH', '/opt/rocm')
+    rocm_lib_path = join(rocm_path, f'lib/{LIBROCM_NAME}')
+    return rocm_lib_path if isfile(rocm_lib_path) else ''
+
+
+def _driver_initialized():
+    """ Returns true if amdgpu is found in the list of initialized modules
+    """
+    initialized = ''
+    try:
+        initialized = str(subprocess.check_output("cat /sys/module/amdgpu/initstate |grep live", shell=True))
+    except subprocess.CalledProcessError:
+        pass
+    return len(initialized) > 0
+
+
+def smi_initialize():
+    """Initialize ROCm binding of SMI"""
+    _load_rocm_library()
+
+    if _driver_initialized():
+        ret_init = rocm_lib.rsmi_init(0)
+        if ret_init != 0:
+            logging.error(f'ROCm SMI init returned value {ret_init}')
```

Comment:
I think this caused test failure.

We shouldn't print error if ROCm fails since we will run it on non-AMD gpus. (Debug log is fine)